CRM/Contact/BAO/Query.php

```diff
@@ -5164,15 +5164,20 @@ public function dateQueryBuilder(
         $date = "'$date'";
       }
 
+      $this->_tables[$tableName] = $this->_whereTables[$tableName] = 1;
+
+      // handle IS NULL / IS NOT NULL / IS EMPTY / IS NOT EMPTY
+      if ($this->nameNullOrEmptyOp($fieldName, $op, $grouping)) {
```

Comment:
Actually it would be better to use _buildClause_ method which intrinsically builds clause based on operator and no extra check etc are needed.